modular_skyrat/modules/delam_emergency_stop/code/scram.dm

```diff
@@ -303,6 +310,19 @@
 /// Proc for arming the red button, it hasn't been pushed yet
 /obj/machinery/button/delam_scram/attack_hand(mob/user, list/modifiers)
 	. = ..()
+
+	if((machine_stat & BROKEN))
+		return
+
+	if(!COOLDOWN_FINISHED(src, scram_button))
+		balloon_alert(user, "on cooldown!")
+		return
+
+	if(!validate_suppression_status())
+		playsound(src.loc, 'sound/machines/buzz-sigh.ogg', 50, FALSE, 10, falloff_distance = BUTTON_SOUND_FALLOFF_DISTANCE)
+		audible_message(span_danger("The [src] makes a sad buzz and goes dark.")) // Look through the window, buddy
+		burn_out()
```

Comment:
Might need a return after this proc call, as I'm guessing you don't want it to continue even after it burnt out.